auslib/web/views/client.py

```diff
@@ -75,22 +75,27 @@ def get(self, **url):
 
             xml = ['<?xml version="1.0"?>']
             xml.append('<updates>')
+            innerXML = []
 
             # We only sample the first blob for the header and footer, since we
             # assume that all blobs will have similar ones. We might want to
             # verify that all of them are indeed the same in the future.
 
-            xml.append(response_blobs[0]['response_release']
-                       .getHeaderXML(response_blobs[0]['product_query'],
-                                     response_blobs[0]['response_update_type']))
+            hasHeaderIfInnerXMLIsEmpty = response_blobs[0]['response_release'].hasHeaderIfInnerXMLIsEmpty()
+
             for response_blob in response_blobs:
-                xml.extend(response_blob['response_release']
-                           .getInnerXML(response_blob['product_query'],
-                                        response_blob['response_update_type'],
-                                        app.config["WHITELISTED_DOMAINS"],
-                                        app.config["SPECIAL_FORCE_HOSTS"]))
-            # Sampling the footer from the first blob
-            xml.append(response_blobs[0]['response_release'].getFooterXML())
+                innerXML.extend(response_blob['response_release']
+                                .getInnerXML(response_blob['product_query'],
+                                             response_blob['response_update_type'],
+                                             app.config["WHITELISTED_DOMAINS"],
+                                             app.config["SPECIAL_FORCE_HOSTS"]))
+            if hasHeaderIfInnerXMLIsEmpty or len(innerXML) != 0:
+                xml.append(response_blobs[0]['response_release']
+                           .getHeaderXML(response_blobs[0]['product_query'],
+                                         response_blobs[0]['response_update_type']))
+                xml.extend(innerXML)
+                # Sampling the footer from the first blob
+                xml.append(response_blobs[0]['response_release'].getFooterXML())
```

Comment:
I think this would be cleaner if you let getHeaderXML/getFooterXML return nothing if uninstall is True instead of adding a new method to the blobs. You can get rid of the need to check for whether or addons exist as well if you use the JSONSchema to ensure that "uninstall" and "vendors" can't be present in the same blob.

What do you think? Is there a reason for this new method that I haven't considered?